### DISCUSSION TITLE
Issue with CSRF token

### DISCUSSION THREAD

**Discussion by timwiebe** [mattieserver/netbox-topology-views#324] [Q&A]
Hello everyone, 
I am using netbox as a docker instance. 
When trying to change the Images in Topology Views I get the following Error Message:

😢 {"detail":"CSRF Failed: CSRF token from the 'X-Csrftoken' HTTP header has incorrect length."}

Any Ideas how I can solve this Issue?

**Comment by mattieserver** [comment]
Can you please specify the version of both netbox and the plugin?

**Reply by timwiebe** [reply]
netbox is on 3.5.3 and plugin on 3.5.0

**Comment by joakim-i** [comment]
Same issue here on version 4.1.0. I am connecting behind a Nginx proxy with the following config:

        proxy_pass https://*****:443;
        proxy_set_header X-Csrftoken $http_x_csrftoken;
        proxy_pass_header X-Csrftoken;
        proxy_set_header Host $host;
        proxy_pass_header Set-Cookie;
        proxy_set_header Cookie $http_cookie;
        proxy_set_header X-Forwarded-Proto https;
        proxy_read_timeout 60;
        proxy_connect_timeout 60;
        proxy_set_header X-Real-IP $remote_addr;
        auth_request /oauth2/auth;
        error_page 401 = /oauth2/start?rd=$request_uri;
        proxy_ssl_verify off;

**Comment by kfgs** [comment]
I'm facing the same problem. I use http only.

In order for **Topology Views** to work correctly in the NetBox Docker image:  

1. Copy the `settings.py` file from the Docker container:  
`docker cp netbox-docker-4_4_1-netbox-1:/opt/netbox/netbox/netbox/settings.py .`  

2. Edit the file and set:  
`CSRF_COOKIE_HTTPONLY = False`  

3. Mount the modified file in your `docker-compose.yml` under the `netbox` service:  
```
services:
  netbox:
    volumes:
      - ./settings.py:/opt/netbox/netbox/netbox/settings.py:ro
```

4. Restart Docker Compose:  
`docker compose down && docker compose up -d`

**Reply by dreng** [reply]
This isn't really a solution but a workaround. The real solution is PR #677.
